util/vram_classes.py

```
@@ -0,0 +1,106 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional
+
+
+@dataclass(frozen=True)
+class VramClass:
+    name: str
+    vram: int
+    given_vram_symbol: Optional[str] = None
+    follows_classes: List[str] = field(default_factory=list, compare=False)
+
+    @property
+    def vram_symbol(self) -> Optional[str]:
+        if self.given_vram_symbol is not None:
+            return self.given_vram_symbol
+        elif self.follows_classes:
+            return self.name + "_CLASS_VRAM"
+        else:
+            return None
+
+
+_vram_classes: Dict[str, VramClass] = {}
+
+
+def initialize(yaml: Any):
+    global _vram_classes
+
+    _vram_classes = {}
+
+    if yaml is None:
+        return
+
+    if not isinstance(yaml, list):
+        raise TypeError("vram_classes must be a list")
```

Why raise an exception instead of using `log.error`?